### PR TITLE
Update all to TypeScript 2.4

### PR DESCRIPTION
**Pull request by staltz**
For issue #640

### CODE REVIEW

**Comment by staltz** [comment]
Could either @jvanbruegge or @Widdershin take a quick look at this?

File: most-run/src/index.ts [hunk 1]
```diff
@@ -51,8 +51,8 @@ export function run<So extends Sources, Si extends Sinks>(
   main: (sources: So) => Si,
   drivers: Drivers<So, Si>,
 ): DisposeFunction {
-  const {run} = coreSetup(main, drivers);
-  return run();
+  const program = coreSetup(main, drivers);
+  return program.run();
```
Comment (jvanbruegge) [inline most-run/src/index.ts]: Why this change?

Comment (staltz) [inline most-run/src/index.ts]: Because we have a new lint rule that warns against shadowing var names. In this case, the local var `run` was conflicting with the module-level exported `run` function.

**Comment by staltz** [comment]
Cool, I'll merge it
